webhook_features/src/features/summary_comment.rs

```diff
@@ -0,0 +1,633 @@
+        url: c.html_url.to_string().clone(),
+        body: c.body.as_ref().unwrap_or(&String::new()).clone(),
+        id: c.id.to_string(),
+    }).collect::<Vec<GitHubReviewComment>>();
+    let mut all_review_comments = ctx.octocrab.all_pages(ctx.octocrab.pulls(&repo.owner, &repo.name)
+        .list_reviews(pr_number).await?)
+        .await?.iter().map(|c| GitHubReviewComment {
+            user: c.user.login.clone(),
+            url: c.html_url.to_string().clone(),
+            body: c.body.as_ref().unwrap_or(&String::new()).clone(),
+            id: c.id.to_string(),
+        }).collect::<Vec<GitHubReviewComment>>();
+
+    all_comments.append(&mut all_review_comments);
+
+
+    let mut ack_per_user: HashMap<String, Vec<AckCommit>> = HashMap::new(); // Need to store all acks per user to avoid duplicates
+    let mut parsed_acks = Vec::new();
+
+    println!("Comments count {}", all_comments.len());
+    for comment in all_comments {
+        // Check comment id is not the same as cmt.id wichi is optional
+        if let Some(cmt_id) = &cmt.id {
+            if cmt_id.to_string() == comment.id {
+                continue;
+            }
+        }
+
+        let acks = parse_acks_in_comment(&comment.body); // A single comment can contain multiple acks, e.g 'Concept ACK and Code Review ACK'
+
+        for ack_commit in acks {
+            let ack_type = ack_commit.ack_type.clone();
+            let commit = ack_commit.commit.clone();
+
+            if commit.is_none() && ack_type.requires_commit_hash() {
+                // If the ack type requires a commit hash, but the comment does not contain one, skip it
+                continue;
+            }
+
+            if ack_per_user.contains_key(&comment.user) // If the user already has an ack for this commit
+                && should_skip_ack(ack_commit.clone(), ack_per_user[&comment.user].clone())
+            // Maybe the ack is a duplicate, need to check for Stale acks
+            {
+                continue;
+            }
+
+            let head = commit.is_some() && commit.as_ref().map_or(false, |c| c.0 == pr.head.sha); // Check if the commit is the head commit of the PR
+
+            parsed_acks.push(Review {
+                user: comment.user.clone(),
+                ack_type,
+                commit: commit.map(|c| (c, head)),
+                url: comment.url.clone(),
+            });
+
+            ack_per_user
+                .entry(comment.user.clone())
+                .or_insert_with(|| vec![ack_commit.clone()]);
+        }
+    }
+
+    let comment = summary_comment_template(false, Some(parsed_acks));
+    util::update_metadata_comment(
+        &ctx.octocrab.issues(&repo.owner, &repo.name),
+        cmt,
+        &comment,
+        util::IdComment::SecReviews,
+        false).await?;
+    Ok(())
+}
+
+#[derive(Clone, Copy, Debug, Hash, PartialEq, Eq)]
+enum AckType {
+    ACK,
+    NACK,
+    CodeReviewACK,
+    ConceptACK,
+    ApproachACK,
+    ApproachNACK,
```

Comment:
It would be really nice if this list is in sync with the guidelines from https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review. If the above stays as it is, do you plan to update `CONTRIBUTING.md` to bring them in sync? Or if the above is to match `CONTRIBUTING.md` then it should be:

```diff
 enum AckType {
     ACK,
-    NACK,
-    CodeReviewACK,
     ConceptACK,
+    ConceptNACK,
     ApproachACK,
     ApproachNACK,
```